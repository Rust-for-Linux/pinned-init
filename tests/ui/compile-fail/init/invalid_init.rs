use pin_init::*;

#[pin_data]
struct Foo {
    #[pin]
    bar: Bar,
}

struct Bar;

impl Bar {
    fn new() -> impl PinInit<Self> {
        Self
    }
}

fn main() {
    let _ = init!(Foo {
        bar <- Bar::new(),
    });
}
