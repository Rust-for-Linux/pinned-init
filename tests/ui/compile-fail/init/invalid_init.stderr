error[E0277]: the trait bound `impl pin_init::PinInit<Bar>: Init<Bar>` is not satisfied
  --> tests/ui/compile-fail/init/invalid_init.rs:18:13
   |
18 |       let _ = init!(Foo {
   |  _____________^
19 | |         bar <- Bar::new(),
20 | |     });
   | |      ^
   | |      |
   | |______the trait `Init<Bar>` is not implemented for `impl pin_init::PinInit<Bar>`
   |        required by a bound introduced by this call
   |
   = help: the trait `Init<T, E>` is implemented for `ChainInit<I, F, T, E>`
   = note: this error originates in the macro `$crate::__init_internal` which comes from the expansion of the macro `init` (in Nightly builds, run with -Z macro-backtrace for more info)
